lib/rubocop/cop/rspec/single_argument_message_chain.rb

```diff
@@ -33,11 +37,27 @@ def on_send(node)
         def autocorrect(node)
           lambda do |corrector|
             corrector.replace(node.loc.selector, replacement(node.method_name))
+            message_chain(node) do |arg|
+              autocorrect_hash_arg(corrector, arg) if single_key_hash?(arg)
+            end
           end
         end
 
         private
 
+        def autocorrect_hash_arg(corrector, arg)
+          key, value = *arg.children.first
+
+          corrector.replace(arg.loc.expression, key_to_arg(key))
+          corrector.insert_after(arg.parent.loc.end,
+                                 ".and_return(#{value.source})")
+        end
+
+        def key_to_arg(node)
+          key, = *node
+          ":#{key.to_sym}"
```

Comment:
Pretty sure this `#to_sym` is unnecessary.

What do you think about

`node.sym_type? ? ":#{key}" : node.source`

to preserve the string or symbol type of keys? I think it would be better if the autocorrect were as true to the original as possible and the choice between string and symbol keys could be left to another cop.